tests/test_elements/test_post_prediction.py

```diff
@@ -0,0 +1,75 @@
+"""Test the module post_prediction.py."""
+
+import unittest
+
+import numpy as np
+from sklearn.base import clone
+from sklearn.decomposition import PCA
+from sklearn.ensemble import RandomForestClassifier
+
+from molpipeline.post_prediction import PostPredictionWrapper
+
+
+class TestPostPredictionWrapper(unittest.TestCase):
+    """Test the PostPredictionWrapper class."""
+
+    def test_get_params(self) -> None:
+        """Test get_params method."""
+        rf = RandomForestClassifier()
+        rf_params = rf.get_params(deep=True)
+
+        ppw = PostPredictionWrapper(rf)
+        ppw_params = ppw.get_params(deep=True)
+
+        wrapped_params = {}
+        for key, value in ppw_params.items():
+            first, sep, rest = key.partition("__")
+            if first == "wrapped_estimator":
+                if rest == "":
+                    self.assertEqual(rf, value)
+                else:
+                    wrapped_params[rest] = value
+
+        self.assertDictEqual(rf_params, wrapped_params)
+
+    def test_set_params(self) -> None:
+        """Test set_params method."""
+        rf = RandomForestClassifier()
+        ppw = PostPredictionWrapper(rf)
+
+        ppw.set_params(wrapped_estimator__n_estimators=10)
+        if not isinstance(ppw.wrapped_estimator, RandomForestClassifier):
+            raise TypeError("Wrapped estimator is not a RandomForestClassifier.")
```

Comment:
I added `self.assertIsInstance(ppw.wrapped_estimator, RandomForestClassifier)`, however I kept the explicit type check, as mypy complains otherwise. (See related [issue](https://github.com/python/mypy/issues/13038)).